### DISCUSSION TITLE
Quick start guide's "User" example type

### DISCUSSION THREAD

**Discussion by rodrigoborgesdeoliveira** [typegoose/typegoose#759] [Q&A] [answered]
Hi, not sure if this is an issue or expected to be like this, so I'm opening a discussion to ask the question. In the [Quick Overview](https://typegoose.github.io/typegoose/docs/guides/quick-start-guide/#quick-overview-of-typegoose) section of the quick start guide, the `User` class is written like this:
```typescript
class User {
  @prop()
  public name?: string;

  @prop({ required: true })
  public age!: number; // This is a single Primitive

  @prop({ type: () => [String] })
  public preferences?: string[]; // This is a Primitive Array

  @prop()
  public mainJob?: Job; // This is a single SubDocument

  @prop({ type: () => Job })
  public jobs?: Job[]; // This is a SubDocument Array

  @prop({ ref: () => Car })
  public mainCar?: Ref<Car>; // This is a single Reference

  @prop({ ref: () => Car })
  public cars?: Ref<Car>[]; // This is a Reference Array
}
```

I was wondering if the `prop` for `jobs` should actually be `@prop({ type: () => [Job] })`. What got me confused is that the `prop` for `preferences` is set as `@prop({ type: () => [String] })`, so I imagined every array would follow that pattern.

**Comment by hasezoey** [comment, marked as answer]
`type: () => [Type]` is a newer syntax than `type: () => Type` (as long as its [`PropType`](https://typegoose.github.io/typegoose/docs/api/decorators/prop/#proptype) is a `ARRAY`) and both work. which style you want to use depends on you. also you can do `type: () => [[Type]]` to create a 2 dimensional array (instead of `type: () => Type, dim: 2` and `type: () => Type` and `type: () => [Type]` both only create a 1 dimensional array

see
- [`dim` option](https://typegoose.github.io/typegoose/docs/api/decorators/prop/#dim)
- [Array Options](https://typegoose.github.io/typegoose/docs/api/decorators/prop/#array-options)

---

thanks for notifying about this, i try to keep the documentation in one style and will (at some point, if i remember) change it to just one style (unless it is meant to explicitly be that style)

**Reply by rodrigoborgesdeoliveira** [reply]
Got it. Thanks for the quick response!
